ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py

```diff
@@ -118,3 +118,36 @@ def test_tensorflow_instance_get_shape(
         fn_tree="Tensor.get_shape",
         self=np.asarray(x, dtype=input_dtype),
     )
+
+
+# sub
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(set(ivy_tf.valid_float_dtypes))
+        ),
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.Tensor.sub",
+    ),
+)
+def test_tensorflow_instance_sub(
+    dtype_and_x, as_variable, num_positional_args, native_array, fw
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_array_instance_method(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        frontend_class=Tensor,
+        fn_tree="Tensor.sub",
+        self=np.asarray(x[0], dtype=input_dtype[0]),
```

Comment:
We have some changes in the policy recently, there is no need to cast the arrays before passing them into the test function now.